src/gui/mainwindow.cpp

```diff
@@ -335,7 +335,7 @@ MainWindow::MainWindow(QWidget *parent)
     // Configure BT session according to options
     loadPreferences(false);
 
-    connect(BitTorrent::Session::instance(), &BitTorrent::Session::torrentsUpdated, this, &MainWindow::updateGUI);
+    connect(BitTorrent::Session::instance(), &BitTorrent::Session::statsUpdated, this, &MainWindow::updateGUI);
```

Comment:
This change is temporary (for qbt to keep working), I plan to clean up `Session::statsUpdated` signal and implement [the idea](https://github.com/qbittorrent/qBittorrent/pull/11147#issuecomment-532688502) in a later PR.

Reply:
Just putting my comment from another thread:
>updateGUI slot has mixed things now (and meaningless name, IMO). It should be split into two methods:
1. Update torrent info. It should depend on the same signals like status filter.
2. Update session statistics. It should depend on statsUpdated signal.

Reply:
>Update torrent info. It should depend on the same signals like status filter.

Also torrent info can be updated only when the appropriate torrent was updated.

Reply:
Not sure if you are waiting this to happen. I would prefer doing it in another PR.

Reply:
Yes, I am.
To finish this job is not so difficult (it requires trivial changes). Do you have a good reason for postponing it?

Reply:
I cannot foresee if it is trivial or it goes down the rabbit hole.

Reply:
Then just not produce the changes that requires this workaround. You can make it all at once "in another PR".

Reply:
reverted